packages/riipen-ui/src/components/RadioGroup.test.jsx

```diff
@@ -0,0 +1,153 @@
+
+    try {
+      mount(<RadioGroup />);
+    } catch (theError) {
+      error = theError;
+    }
+
+    expect(error).toEqual(undefined);
+  });
+
+  it("renders correct snapshot", () => {
+    const wrapper = mount(<RadioGroup />);
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  describe("children prop", () => {
+    it("renders given non null children", () => {
+      const child = <Radio label="one" readOnly />;
+      const child2 = <Radio label="two" readOnly />;
+
+      const wrapper = mount(
+        <RadioGroup>
+          {child}
+          {null}
+          {child2}
+        </RadioGroup>
+      );
+
+      const childrenNodes = wrapper.find("fieldset").find("Radio");
+      expect(childrenNodes.at(0).props().label).toEqual("one");
+      expect(childrenNodes.at(1).props().label).toEqual("two");
+    });
+  });
+
+  describe("error prop", () => {
+    it("displays error", () => {
+      const error = <span>an error!</span>;
+
+      const wrapper = mount(<RadioGroup error={error} />);
+
+      expect(wrapper.find("fieldset").contains(error)).toEqual(true);
+    });
+  });
+
+  describe("hint prop", () => {
+    it("renders InputLabel when hint is provided", () => {
+      const hint = "it's a hint";
+
+      const wrapper = mount(<RadioGroup hint={hint} />);
+
+      expect(wrapper.find("InputLabel")).toHaveLength(1);
+    });
+
+    it("passes given hint to InputLabel", () => {
+      const hint = "it's a hint";
+
+      const wrapper = mount(<RadioGroup hint={hint} />);
+
+      expect(wrapper.find("InputLabel").props().hint).toEqual(hint);
+    });
+  });
+
+  describe("label prop", () => {
+    it("renders InputLabel when label is provided", () => {
+      const label = "it's a label";
+
+      const wrapper = mount(<RadioGroup label={label} />);
+
+      expect(wrapper.find("InputLabel")).toHaveLength(1);
+    });
+
+    it("passes given label to InputLabel as children", () => {
+      const label = "it's a label";
+
+      const wrapper = mount(<RadioGroup label={label} />);
+
+      expect(wrapper.find("InputLabel").props().children).toEqual(label);
+    });
+
+    it("does not render InputLabel when both label and hint are not provided", () => {
+      const wrapper = mount(<RadioGroup />);
+
+      expect(wrapper.find("InputLabel")).toHaveLength(0);
+    });
+  });
+
+  describe("onChange prop", () => {
+    it("sets onChange prop of children nodes", () => {
+      const child = <Radio label="one" readOnly />;
+
+      const wrapper = mount(<RadioGroup>{child}</RadioGroup>);
+
+      expect(wrapper.find("Radio").props().onChange).toBeDefined();
+    });
+
+    it("invokes handler when child Radio is changed", () => {
+      const child = <Radio label="one" readOnly />;
+      const onChange = jest.fn();
+
+      const wrapper = mount(
+        <RadioGroup onChange={onChange}>{child}</RadioGroup>
+      );
+
+      wrapper
+        .find("Radio")
+        .find("input")
+        .simulate("change");
+
```

Comment:
Should there also be a test that covers the cloneElement function in lines 72-73 in RadioGroup? 
ie. something like...
expect(wrapper.find("fieldset").contains(childrenWithProps)).toEqual(true);

Reply:
I think we are already testing that with the `children` tests.